### PR TITLE
Extract JsonSchemaProvider from TemplateEngine

### PR DESCRIPTION
**Pull request by AngryBerryMS**

### CODE REVIEW

File: src/docfx/template/TemplateSchemaProvider.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        }
+
+        public JsonSchemaValidator GetSchemaValidator(SourceInfo<string?> mime)
+        {
+            var name = mime.Value ?? throw Errors.Yaml.SchemaNotFound(mime).ToException();
+
+            return _schemas.GetOrAdd(name, GetSchemaCore) ?? throw Errors.Yaml.SchemaNotFound(mime).ToException();
+        }
+
+        public string? GetToken(string key)
```
Comment (yufeih) [inline src/docfx/template/TemplateSchemaProvider.cs:46]: This belongs to `TemplateEngine`.

File: src/docfx/lib/TemplateEngineUtility.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        public static bool IsConceptual(string? mime) => "Conceptual".Equals(mime, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsLandingData(string? mime) => "LandingData".Equals(mime, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsMigratedFromMarkdown(string? mime)
+        {
+            return mime != null && s_yamlMimesMigratedFromMarkdown.Contains(mime);
+        }
+
+        public static JObject LoadGlobalTokens(ErrorBuilder errors, Package package, string locale)
```
Comment (yufeih) [inline src/docfx/lib/TemplateEngineUtility.cs:38]: Why move these methods outside `TemplateEngine`?

File: src/docfx/lib/TemplateEngineUtility.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            "LandingData",
+        };
+
+        public static bool OutputAbsoluteUrl(string? mime) => mime != null && s_outputAbsoluteUrlYamlMime.Contains(mime);
+
+        public static bool IsConceptual(string? mime) => "Conceptual".Equals(mime, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsLandingData(string? mime) => "LandingData".Equals(mime, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsMigratedFromMarkdown(string? mime)
```
Comment (yufeih) [inline src/docfx/lib/TemplateEngineUtility.cs:33]: These mime type related static method feels more related to `JsonSchemaProvider`

File: src/docfx/template/TemplateSchemaProvider.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class TemplateSchemaProvider
```
[the remark this answers is not in the document]
Comment (AngryBerryMS) [reply src/docfx/template/TemplateSchemaProvider.cs]: Fixed.

File: src/docfx/lib/TemplateEngineUtility.cs (hunk 2 above)
Comment (AngryBerryMS) [reply src/docfx/lib/TemplateEngineUtility.cs]: Fixed

File: src/docfx/template/TemplateSchemaProvider.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/template/TemplateSchemaProvider.cs]: Fixed.

File: src/docfx/lib/TemplateEngineUtility.cs (hunk 3 above)
Comment (AngryBerryMS) [reply src/docfx/lib/TemplateEngineUtility.cs]: Fixed.

**Review by yufeih — APPROVED** [review]

**Pull request merged**